tests/PySys/monitoring/monitoring_with_collectd/run.py

```diff
@@ -0,0 +1,124 @@
+from pysys.basetest import BaseTest
+
+import time
+import re
+import json
+
+"""
+Validate tedge-dm-agent  messages that are published
+on tedge/measurements
+
+Given a configured system
+When we start the collectd with sudo in the background
+When we start the tedge-dm-agent with sudo in the background
+When we start tedge sub with sudo in the background
+Wait for couple of seconds to publish couple of batch of messages
+Then we kill tedge sub with sudo as it is running with a different user account
+Then we validate the  messages in the output of tedge sub,
+"""
+
+
+class MonitoringWithCollectd(BaseTest):
+
+    def setup(self):
+        self.js_msg = ""
+        self.cpu_cnt = 0
+        self.memory_cnt = 0
+        self.time_cnt = 0
+        self.disk_cnt = 0
+
+    def execute(self):
+        tedge = "/usr/bin/tedge"
+        sudo = "/usr/bin/sudo"
+
+        collectd = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "collectd"],
+            stdouterr="collectd",
+        )
+
+        collectd_mapper = self.startProcess(
+            command=sudo,
+            arguments=["systemctl", "start", "tedge-dm-agent"],
+            stdouterr="collectd_mapper",
+        )
+
+        time.sleep(0.1)
+
+        sub = self.startProcess(
+            command=sudo,
+            arguments=[tedge, "mqtt", "sub", "--no-topic", "tedge/#"],
+            stdouterr="tedge_sub",
+            background=True,
+        )
+
```

Comment:
I'd suggest that you start another `tedge mqtt sub collectd/#` command here just for debugging purposed so that in future, if the test fails, we'll be able to compare the `collectd/#` output with the `tedge/#` output. You don't need to do any validation on the output of that command. It's just there for future debugging.

Reply:
ok, I doubt this is good to be part of the test because we never use this in the test. I am ok to add one more sub for  collecting the collectd logs.